chore: lock all package versions to match yarn.lock

volnei opened the pull request:
## What does this PR do?

Pins all dependency versions in package.json files to exact versions matching the yarn.lock file by removing `^` and `~` prefixes. This ensures reproducible builds and prevents unexpected version drift.

**Summary:**
- Locked 427+ dependencies across 47+ package.json files
- Versions now match exactly what is resolved in yarn.lock
- Used `yarn why` approach to determine the correct locked versions

### Updates since last revision

Addressed PR review comments:
- **Prisma version alignment**: Upgraded `@prisma/client` from `6.7.0` to `6.16.1` in example apps (`example-apps/credential-sync` and `packages/platform/examples/base`) to match the main app's Prisma version
- **Additional pinning**: Pinned `@types/node` to `20.17.23` in `packages/platform/atoms`, `packages/platform/libraries`, and example apps
- **Type definitions**: Pinned `@types/react` to `18.0.26` and `@types/react-dom` to `18.2.6` in example apps
- **yarn.lock updated**: Committed updated yarn.lock reflecting the Prisma version upgrade and descriptor key changes

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works. N/A - no functional changes, only version pinning.
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a documentation change. N/A

## How should this be tested?

1. Run `yarn install` to verify dependencies resolve correctly
2. Run `yarn lint` and `yarn type-check:ci` to verify no regressions
3. Spot check a few package.json files against yarn.lock to verify versions match

## Human Review Checklist

- [ ] Verify Prisma versions are aligned: `@prisma/client` and `prisma` should both be `6.16.1` in example apps
- [ ] Note that peerDependencies intentionally retain version ranges (e.g., `react: "^18.0.0 || ^19.0.0"`) for compatibility
- [ ] Some versions changed beyond just removing the prefix (e.g., `@googleapis/admin: ^23.0.0 → 23.5.0`) because yarn.lock had resolved to newer versions within the semver range
- [ ] Verify yarn.lock changes are primarily descriptor key rewrites (e.g., `@pkg@npm:^1.0.0` → `@pkg@npm:1.0.0`) rather than resolution changes
- [ ] Confirm this aligns with the team's dependency management strategy

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have checked if my changes generate no new warnings

---

**Link to Devin run:** https://app.devin.ai/sessions/46509824870646afb1a08084ea843d0b
**Requested by:** Volnei Munhoz (@volnei)

devin-ai-integration[bot] committed:
chore: lock all package versions to match yarn.lock

This commit pins all dependency versions in package.json files to exact
versions matching the yarn.lock file, removing ^ and ~ prefixes.

Changes:
- Locked 427 dependencies across 47 package.json files
- Versions now match exactly what is resolved in yarn.lock
- Ensures reproducible builds and prevents unexpected version drift

This change improves build reproducibility by ensuring that the versions
specified in package.json files match exactly what yarn.lock resolves to.

Co-Authored-By: Volnei Munhoz <volnei.munhoz@gmail.com>

volnei committed:
Merge branch 'main' into devin/1766749050-lock-package-versions

volnei committed:
Merge branch 'main' into devin/1766749050-lock-package-versions

keithwillcode commented on packages/platform/examples/base/package.json:
```diff
@@ -15,21 +15,21 @@
     "@prisma/client": "6.7.0",
     "next": "16.1.0",
-    "prisma": "^6.7.0",
+    "prisma": "6.16.1",
```
@volnei seems we should address

keithwillcode reviewed (commented):
Looking good. Cubic pointed out some things around prisma. What do you think?

devin-ai-integration[bot] committed:
fix: upgrade Prisma versions in example apps and pin remaining @types/node

- Upgraded @prisma/client to 6.16.1 in example apps to match main app
- Pinned @types/node to 20.17.23 in atoms, libraries, and example apps
- Pinned @types/react to 18.0.26 and @types/react-dom to 18.2.6 in example apps

Addresses PR review comments about Prisma version mismatch and unpinned @types/node

Co-Authored-By: Volnei Munhoz <volnei.munhoz@gmail.com>

devin-ai-integration[bot] committed:
chore: update yarn.lock after Prisma version upgrade

Updates yarn.lock to reflect:
- @prisma/client upgraded from 6.7.0 to 6.16.1 in example apps
- Removed unused @prisma/client@6.7.0 resolution

Co-Authored-By: Volnei Munhoz <volnei.munhoz@gmail.com>
